src/Confluent.SchemaRegistry/Rest/RestService.cs

```diff
@@ -52,23 +53,29 @@ internal class RestService : IRestService
         ///     Initializes a new instance of the RestService class.
         /// </summary>
         public RestService(string schemaRegistryUrl, int timeoutMs, string username, string password)
-        { 
-            var authorizationHeader = username != null && password != null 
+        {
+            var authorizationHeader = username != null && password != null
                 ? new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.UTF8.GetBytes($"{username}:{password}")))
                 : null;
 
             this.clients = schemaRegistryUrl
                 .Split(',')
-                .Select(uri => uri.StartsWith("http", StringComparison.Ordinal) ? uri : "http://" + uri) // need http or https - use http if not present.
-                .Select(uri => 
-                    {
-                        var client = new HttpClient() { BaseAddress = new Uri(uri, UriKind.Absolute), Timeout = TimeSpan.FromMilliseconds(timeoutMs) };
-                        if (authorizationHeader != null) { client.DefaultRequestHeaders.Authorization = authorizationHeader; }
-                        return client;
-                    })
+                .Select(SanitizeUri)// need http or https - use http if not present.
+                .Select(uri =>
+                {
+                    var client = new HttpClient { BaseAddress = new Uri(uri, UriKind.Absolute), Timeout = TimeSpan.FromMilliseconds(timeoutMs) };
+                    if (authorizationHeader != null) { client.DefaultRequestHeaders.Authorization = authorizationHeader; }
+                    return client;
+                })
                 .ToList();
         }
 
+        private static string SanitizeUri(string uri)
+        {
+            var sanitized = uri.StartsWith("http", StringComparison.Ordinal) ? uri : $"http://{uri}";
+            return $"{Regex.Replace(sanitized, "(/$)+", string.Empty)}/";
```

Comment:
can you change this to not use `System.Text.RegularExpressions`? i think this might result in an assembly being loaded that otherwise wouldn't be, and the logic's simple enough that using a regex doesn't add much.

Reply:
done